### ISSUE TITLE
FeeInsufficient

### ISSUE THREAD

**Issue by robtex** [lightningnetwork/lnd#1549]
trying to pay thru a specific path created by queryroutes gives FeeInsufficient
i don't know if it is my lnd that calculates the fee or the intermediate one.
the intermediate node 03a2b9adc3086b0ba7844bcda0159f11967e5558e430f168ee7ee797cb9830d742
has the alias `scumm-lnd` thus i suspect it is also a lnd

made queryroutes and extracted a specific path

```lncli queryroutes 036a33e92cda2679ac91ac0467acb1ba33396c0df1119521653e0f269c275dd6bf 1000  --num_max_routes 100 --final_cltv_delta 1000```

this route is the interesting one

savedroute:
```
{
        "routes":
        [{
            "total_time_lock": 532746,
            "total_fees": "0",
            "total_amt": "1000",
            "hops": [
                {
                    "chan_id": "580442083918675968",
                    "chan_capacity": "47187",
                    "amt_to_forward": "1000",
                    "fee": "0",
                    "expiry": 532602,
                    "amt_to_forward_msat": "1000000",
                    "fee_msat": "1"
                },
                {
                    "chan_id": "580679578471170048",
                    "chan_capacity": "300000",
                    "amt_to_forward": "1000",
                    "fee": "0",
                    "expiry": 532602,
                    "amt_to_forward_msat": "1000000",
                    "fee_msat": "0"
                }
            ],
            "total_fees_msat": "1",
            "total_amt_msat": "1000001"
        }]
}
```

```
lncli sendtoroute --payment_hash=6a33e92cda2679ac91ac0467acb1ba33396c0df1119521653e0f269c275dd6bf - <savedroute
```

```
{
        "payment_error": "unable to route payment to destination: FeeInsufficient(fee=1000000 mSAT, update=(lnwire.ChannelUpdate) {\n Signature: (lnwire.Sig) (len=64 cap=64) {\n  00000000  9e 6f 17 46 c2 bb 5f a5  2b a0 82 d6 89 c5 68 51  |.o.F.._.+.....hQ|\n  00000010  77 cc 24 53 23 e5 e3 33  8b 1f 0e 38 ed c3 67 94  |w.$S#..3...8..g.|\n  00000020  03 24 a0 db a7 1a cb 39  f6 99 11 66 d1 47 9e 24  |.$.....9...f.G.$|\n  00000030  e2 f0 67 ec af 6d e9 d2  09 0e d1 a5 00 e7 02 78  |..g..m.........x|\n },\n ChainHash: (chainhash.Hash) (len=32 cap=32) 000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f,\n ShortChannelID: (lnwire.ShortChannelID) 528125:793:0,\n Timestamp: (uint32) 1531353087,\n Flags: (lnwire.ChanUpdateFlag) 0,\n TimeLockDelta: (uint16) 144,\n HtlcMinimumMsat: (lnwire.MilliSatoshi) 1000 mSAT,\n BaseFee: (uint32) 1000,\n FeeRate: (uint32) 1\n}\n",
        "payment_preimage": "",
        "payment_route": null
}
```

channel 580442083918675968:
```
        {
            "channel_id": "580442083918675968",
            "chan_point": "acd382d13bbde40476f793f3285a43a32b5aee13bc0b3ba504ff085c9c168533:0",
            "last_update": 1531393098,
            "node1_pub": "02ad6fb8d693dc1e4569bcedefadf5f72a931ae027dc0f0c544b34c1c6f3b9a02b",
            "node2_pub": "03a2b9adc3086b0ba7844bcda0159f11967e5558e430f168ee7ee797cb9830d742",
            "capacity": "100000",
            "node1_policy": {
                "time_lock_delta": 144,
                "min_htlc": "1000",
                "fee_base_msat": "0",
                "fee_rate_milli_msat": "1"
            },
            "node2_policy": {
                "time_lock_delta": 144,
                "min_htlc": "1000",
                "fee_base_msat": "0",
                "fee_rate_milli_msat": "1"
            }
        },
```

channel:580679578471170048
```
       {
            "channel_id": "580679578471170048",
            "chan_point": "920853c333a11d268b8162b31c755592394ef185a53bfa4d4bf2226ff66e7d98:0",
            "last_update": 1531353087,
            "node1_pub": "036a33e92cda2679ac91ac0467acb1ba33396c0df1119521653e0f269c275dd6bf",
            "node2_pub": "03a2b9adc3086b0ba7844bcda0159f11967e5558e430f168ee7ee797cb9830d742",
            "capacity": "300000",
            "node1_policy": {
                "time_lock_delta": 144,
                "min_htlc": "1000",
                "fee_base_msat": "1000",
                "fee_rate_milli_msat": "1"
            },
            "node2_policy": {
                "time_lock_delta": 144,
                "min_htlc": "1000",
                "fee_base_msat": "0",
                "fee_rate_milli_msat": "1"
            }
        },
```

### Background

Describe your issue here.

### Your environment

* version of `lnd`
* which operating system (`uname -a` on *Nix)
* version of `btcd`, `bitcoind`, or other backend
* any other relevant environment details

### Steps to reproduce

Tell us how to reproduce this issue. Please provide stacktraces and links to code in question.

### Expected behaviour

Tell us what should happen

### Actual behaviour

Tell us what happens instead

**Comment by halseth** [comment]
What commit are you on? Might be fixed by  #1382 or #1321.

**Comment by robtex** [comment]
Yeah one of those probably might fix it.
Rats... I investigated how queryroutes calculated fees and thought it looked a bit strange, but i used it's output as a testcase for my fee calculator in my channel balancer...
so i guess i successfully emulated the bug too.
I'll check back later.

**Comment by Roasbeef** [comment]
Which commit are you on? Also note that if you're routing through peers that haven't been updated then they'll give you this error.

**Comment by robtex** [comment]
just updated to latest master on two of lnd:s under my control to see if i can replicated it or if it happens by itself.
previous nodes was about a week old commit, but the scumm-lnd node in this issue i have no idea who it is so can't ask what version anyhows

**Comment by halseth** [comment]
Note that `queryroutes` has its own path finding logic, separate from `missioncontrol`. This means that it still potentially has the bugs fixed by #1382 and #1321.

**Comment by robtex** [comment]
@halseth thanks. yes it still looks fishy and i just did the same test again (from master without #1321  ) with same result.
duplicated code is awesome
i think i'll refactor my own path finding logic and using trial and error + my own brain instead.
i assume `sendtoroute`:s handling of the routes json won't be changed, right?

**Comment by halseth** [comment]
Making P3, as this seems to apply to `queryRoutes` only. It should also get the path finding fixes done by #1321.

**Comment by Roasbeef** [comment]
Fixed by #1321.
